src/compiler/types/tests/generate-app-types.spec.ts

```diff
@@ -1406,6 +1409,198 @@ declare module "@stencil/core" {
                 }
         }
 }
+`,
+      {
+        immediateWrite: true,
+      }
+    );
+  });
+
+  it('should transform aliased paths if the config option is true', async () => {
+    const compilerComponentMeta = stubComponentCompilerMeta({
+      tagName: 'my-component',
+      componentClassName: 'MyComponent',
+      jsFilePath: path.join(config.rootDir, 'some/stubbed/path/a/my-component.js'),
+      sourceFilePath: path.join(config.rootDir, 'some/stubbed/path/a/my-component.tsx'),
+      sourceMapPath: path.join(config.rootDir, 'some/stubbed/path/a/my-component.js.map'),
+      hasProp: true,
+      properties: [
+        stubComponentCompilerProperty({
+          name: 'name',
+          complexType: {
+            original: 'UserImplementedPropType',
+            resolved: '"foo" | "bar"',
+            references: {
+              UserImplementedPropType: {
+                location: 'import',
+                path: '@utils',
+              },
+            },
+          },
+        }),
+      ],
+    });
+    buildCtx.components = [compilerComponentMeta];
+    config.tsCompilerOptions = {
+      paths: {
+        '@utils': [path.join(config.rootDir, 'some/stubbed/path/utils/utils.ts')],
+      },
+      declaration: true,
+    };
+    config.transformAliasedImportPaths = true;
+    // We need to have a file in the in-memory fs for the TS module resolution to succeed
+    originalWriteFile(path.join(config.rootDir, 'some/stubbed/path/utils/utils.ts'), '');
+
+    patchTypescript(config, compilerCtx.fs);
```

Comment:
Can we 'unpatch' this in between tests?

Reply:
Easily? I don't think so. Looks like that utility overwrites a few methods on the TS object, but it saves the originals on properties we create (like `__resolveModuleName`). So, we _could_ create an `unpatchTypescript` method that reassigns the original definitions to their signatures.

Reply:
Yeah, you're right - let's punt on this for now